### ISSUE TITLE
False positive in C416

### ISSUE THREAD

**Issue by NoAnyLove** [adamchainz/flake8-comprehensions#196]
Example file, `demo.py`
```python
import asyncio


async def ticker(delay, to):
    for i in range(to):
        yield i
        await asyncio.sleep(delay)


async def main():
    return [item async for item in ticker(1, 10)]

```

Error message,
```
demo.py:5:15: C416 Unnecessary list comprehension - rewrite using list()
```

I don't think we can rewrite asynchronous list/dict/set comprehensions (https://www.python.org/dev/peps/pep-0525/#asynchronous-list-dict-set-comprehensions) using `list()`, `dict()` or `set()`.

**Comment by adamchainz** [comment]
Fixed in version 3.1.1. https://pypi.org/project/flake8-comprehensions/3.1.1/

**Comment by adamchainz** [comment]
I scanned the other rules but couldn't spot any obvious places where there'd be a false positive for async comprehensions, if there are please let me know / make PR.

**Comment by NoAnyLove** [comment]
I think `is_async` is also available in Python 3.6,
```
$ python
Python 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ast
>>> 
>>> code='async def demo():\n   [i async for i in dummy]'
>>> ast.parse(code).body[0].body[0].value.generators[0]
<_ast.comprehension object at 0x7fb65be13550>
>>> ast.parse(code).body[0].body[0].value.generators[0].is_async
1
>>> 
```

**Comment by adamchainz** [comment]
Right - the problem was my tests were using `async for` outside of an `async` function, which is only valid in 3.7+. Fixed in #202, released in 3.1.3: https://pypi.org/project/flake8-comprehensions/3.1.3/
